js/wee.history.js

```diff
@@ -367,6 +389,15 @@
 			request.url = url;
 			conf.request = request;
 
+			if (conf.begin && W.$exec(conf.begin, {
```

Comment:
@leigeber is there a particular reason that `replaceState` needs to occur before the request url is set? I have recently found advantages to executing conf.begin before `replaceState`. A specific case was working with Google Tag Manager. Because Wee.history.go executes `replaceState` and `pushState`, the GTM history trigger cannot accurately be used since it listens for both of these methods. A hack for this is to set a global variable in `begin` and then reset that same variable in a lifecycle method proceeding `replaceState` in order to inform GTM to only trigger on `pushState`. In general, it makes sense to have `begin` as early as possible. Thoughts?